oci/config.py

```diff
@@ -1,46 +1,46 @@
 from collections import namedtuple
-import ConfigParser
 import os
+from six.moves import configparser
 
-gerrit_default_host = 'gerrit.ovirt.org'
-jenkins_default_host = 'jenkins.ovirt.org'
+DEFAULT_CONFIG = {
+    ('jenkins', 'host', 'jenkins.ovirt.org'),
+    ('gerrit', 'host', 'gerrit.ovirt.org')}
 
+Jenkins = namedtuple('jenkins', 'host, user_name, api_token')
+Gerrit = namedtuple('gerrit', 'host')
+Config = namedtuple('config', 'jenkins, gerrit')
 
-def load():
-    config_dir = os.path.join(os.path.expanduser('~'), '.config/oci.conf')
-    config = ConfigParser.RawConfigParser(allow_no_value=True)
 
-    with open(config_dir, 'r') as config_file:
-        config.readfp(config_file)
+class Error(Exception):
+    pass
 
-        # set default Jenkins host if not present
-        jenkins_host = jenkins_default_host
-        if config.has_option('jenkins', 'host'):
-            jenkins_host = config.get('jenkins', 'host')
 
-        # set default Gerrit host if not present
-        gerrit_host = jenkins_default_host
-        if config.has_option('gerrit', 'host'):
-            gerrit_host = config.get('gerrit', 'host')
+def config_parser():
+    cfg = configparser.RawConfigParser()
 
+    # Setup default configuration
+    for (section, key, value) in DEFAULT_CONFIG:
+        cfg.add_section(section)
+        cfg.set(section, key, value)
+
+    return cfg
+
+
+def load(path=os.path.expanduser("~/.config/oci.conf")):
+    cfg = config_parser()
+
+    with open(path, 'r') as config_file:
+        cfg.readfp(config_file)
         try:
-            token = config.get('jenkins', 'api_token')
-            user = config.get('jenkins', 'user_id')
-        except ConfigParser.NoOptionError as e:
-            raise Exception(
-                'A setting in the config file is missing: ', e)
-
-    Jenkins = namedtuple('jenkins', 'host, user_name, api_token')
-    Gerrit = namedtuple('gerrit', 'host')
-    Config = namedtuple('config', 'jenkins, gerrit')
-
-    result = Config(
-        jenkins=Jenkins(
-            host=jenkins_host,
-            user_name=user,
-            api_token=token),
-        gerrit=Gerrit(
-            host=gerrit_host
-        ))
-
-    return result
+            return Config(
+                jenkins=Jenkins(
+                    host=cfg.get('jenkins', 'host'),
+                    user_name=cfg.get('jenkins', 'user_id'),
+                    api_token=cfg.get('jenkins', 'api_token')),
+                gerrit=Gerrit(
+                    host=cfg.get('gerrit', 'host')
+                ))
+        except configparser.NoOptionError as err:
+            raise Error(
+                "No option '%s' in section '%s' in OCI config file."
+                % (err.option, err.section))
```

Comment:
Better to use .format() when we can:

```python
    raise Error(
        "Option {!r} in section {!r} is required".format(err.option, err.section))
```

Also note that we can use formatting codes instead of manual quoting.